### PR TITLE
Fix flaky @babel/cli test

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | The `@babel/cli` watch test is flaky because lack of inter-process communication
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14385"></a>

To re-run the `@babel/cli` test again and again, I have pushed some CI maintenance commits.

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
I love this PR

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,16 +27,16 @@ jobs:
...
-        uses: actions/setup-node@v2-beta
+        uses: actions/setup-node@v3
         with:
           node-version: "*"
+          cache: "yarn"
       - name: 'Check for unmet constraints (fix w/ "yarn constraints --fix")'
         run: |
           yarn constraints
       - name: 'Check for duplicate dependencies (fix w/ "yarn dedupe")'
-        if: steps.yarn-cache.outputs.cache-hit != 'true'
```
Comment (nicolo-ribaudo) [inline .github/workflows/ci.yml:48]: We do we always run it now?

**Review by nicolo-ribaudo — APPROVED** [review]
This _might_ still not solve it because it just "waits more", but 2s should be more then enough :+1:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JLHwung) [reply .github/workflows/ci.yml:48]: Good catch. I should have left a comment here. This step was previously in the `prepare-yarn-cache` job including the step `yarn-cache`. Not long ago it was extracted to a separate job `yarn-validate` which can't access the `yarn-cache` output anymore, so the condition here is always `true` now. The dedupe check is around 2 seconds so we can just leave it here.

**Comment by nicolo-ribaudo** [comment]
I'm merging this with a single review because it doesn't affect any source code, and it fixes a super annoying problem that makes ~30% of CI runs fail.

**Comment by liuxingbaoyu** [comment]
There seems to be a failure for the same reason.

We can consider adding a watch ready output.

Or consider the following changes:

```js
  await new Promise(resolve => setTimeout(resolve, 2000));
  fs.writeFileSync("./file.txt", "Updated!");
```

```js
 let i = 0;
  setInterval(() => {
    fs.writeFileSync("./file.txt", `Updated${i++}!`);
  }, 300);
```

**Pull request merged**